proposals/XXXX-compact.md

```diff
@@ -96,79 +104,167 @@ Compact witness type signatures are a kind of foreign function interface (FFI).
+The Midnight Network has a process for proposing changes called the Midnight Improvement Proposal (MIP) process.
+The Compact project will use this process or a suitable variation of it to conduct language feature design in the open,
+and to allow community-driven language feature design.
 
-<mark>_**Effort and resources** committed (coders and any other resources
-that are needed) and timeline._
+An example of a similar process being used successfully for programming languages can be found in the Python language community's
+[Python Enhancement Proposal (PEP)](https://peps.python.org/) process.
+
+<mark>**Effort and resources** committed (coders and any other resources that are needed) and timeline.
 </mark>
 
 # How To
-<mark>_**How to**: How to host and test the project. How to deploy and use.
-How does one know that it works._
+<mark>**How to**: How to host and test the project.
+How to deploy and use.
+How does one know that it works.
 </mark>
 
 # References
-<mark>_**References**. See [citation guide](http://www.chicagomanualofstyle.org/tools_citationguide.html)._
+<mark>**References**. See [citation guide](http://www.chicagomanualofstyle.org/tools_citationguide.html).
 </mark>
 
-# Closure
-<mark>_**Closure** how do we know that the project succeeded. This has to
-be measurable if possible. Make references to successor projects if
-any._
-</mark>
+TODO, suggestions:
+- Chez Scheme or TSPL?
+- Nanopass framework?
+- MIP repository?
+- The long Kachina paper?
+
+# Closure: Success Criteria for Compact
+The Compact project will be considered successfully completed when the following measurable criteria have been met.
+These criteria are organized by the major deliverables and goals of the project.
+
+1. **Language Specification and Documentation**
+
+   The goal is to have a clear, unambiguous, and complete definition of the language that can be used by developers, auditors, and implementers.
+
+   1. **Language Reference Completeness:** The informal Compact Language Reference document fully describes 100% of the language features implemented in the compiler and used in the Standard Library.
+      There are no compiler features that lack corresponding documentation.
+
+   1. **Specification Consistency:** An external review by at least two developers (not on the core Compact team) confirms they can build a functionally equivalent "hello world" style contract using only the Language Reference, with no ambiguities found.
+
+   1. **Standard Library Documentation:** Every function and module in the Compact Standard Library is documented with its type signature, purpose, parameters, and a usage example.
+1. **Compiler and Software Development Kit (SDK)**
+
+   The goal is to provide a robust, correct, and usable compiler and set of tools for developers.
+
+   1. **Compiler Correctness and Test Coverage:** The compiler's test suite achieves a minimum of 90% code coverage.
```

Comment:
We're currently at 88% and there isn't much more we can pick by feeding the compiler Compact programs.  There is a lot of unreachable code in the three downstream type-checking passes that we can only test if we feed them bogus intermediate-language programs.  I'd rather not do that.  Perhaps say the goal is 90% excluding self-diagnostic code that is exercised only when the compiler itself is broken.